pkg/azure/provider/helpers/driver.go

```diff
@@ -115,11 +116,23 @@ func GetDiskNames(providerSpec api.AzureProviderSpec, vmName string) []string {
 	dataDisks := providerSpec.Properties.StorageProfile.DataDisks
 	diskNames := make([]string, 0, len(dataDisks)+1)
 	diskNames = append(diskNames, utils.CreateOSDiskName(vmName))
-	if !utils.IsSliceNilOrEmpty(dataDisks) {
-		for _, disk := range dataDisks {
-			diskName := utils.CreateDataDiskName(vmName, disk)
-			diskNames = append(diskNames, diskName)
-		}
+	dataDiskNames := getDataDiskNames(providerSpec, vmName)
+	if dataDiskNames != nil {
+		diskNames = append(diskNames, dataDiskNames...)
+	}
+	return diskNames
+}
+
+// getDataDiskNames creates disk names for all configured DataDisks in the provider spec.
+func getDataDiskNames(providerSpec api.AzureProviderSpec, vmName string) []string {
+	dataDisks := providerSpec.Properties.StorageProfile.DataDisks
+	if utils.IsSliceNilOrEmpty(dataDisks) {
+		return nil
+	}
+	diskNames := make([]string, 0, len(dataDisks)+1)
```

Comment:
Is there a need to have the `len != len(dataDisks)` instead of `len(dataDisks) +1`